### ISSUE TITLE
Issue with PHP 7.4

### ISSUE THREAD

**Issue by AntoineLemaire** [doctrine/persistence#154]
Hello,

We're trying to migrate our project to PHP 7.4 and we face a issue

```
PHP Warning:  Invalid argument supplied for foreach() in /var/www/myproject/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php on line 179
```

I was able to create a test project which you can reproduce it. (Sorry, it's an old version of Symfony, I tried to keep same structure of our production project)

```
git clone git@github.com:AntoineLemaire/doctrine-annotation-issue.git
cd doctrine-annotation-issue
composer install --optimize-autoloader --no-progress --ansi --prefer-dist --no-suggest
```

Then, when you do several times `app/console cache:warmup --no-interaction --env=dev --no-debug`, after few tries you got this error

![image](https://user-images.githubusercontent.com/2025537/104932135-99d1b500-59a7-11eb-9ede-808ca734e794.png)

I tried with the exact same project & PHP 7.3, and I didn't had any issue.

Here the stack:
- symfony/symfony 3.4.20 (we're stuck at this version at the moment)
- PHP 7.4.13
- Composer 1.10.8
- doctrine/annotations 1.11.1

**Comment by greg0ire** [comment]
Hi! What version of `doctrine/persistence` are you using?
Also, please provide a stack trace, you can read more here about how to obtain one when using the CLI: https://symfony.com/doc/current/contributing/code/stack_trace.html#stack-traces-in-the-cli

**Comment by AntoineLemaire** [comment]
Hi! We are on `doctrine/persistence` 1.3.8
I commited the [composer.lock](https://github.com/AntoineLemaire/doctrine-annotation-issue/blob/master/composer.lock)

Here the stack trace:

```php
08:56:42 WARNING   [php] Warning: Invalid argument supplied for foreach()
[
  "exception" => ErrorException {
    #message: "Warning: Invalid argument supplied for foreach()"
    #code: 0
    #file: "./vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php"
    #line: 179
    #severity: E_WARNING
    trace: {
      ./vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php:179 { …}
      ./vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php:245 { …}
      ./vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/MappingDriverChain.php:107 { …}
      ./vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php:90 { …}
      ./vendor/symfony/symfony/src/Symfony/Bridge/Doctrine/CacheWarmer/ProxyCacheWarmer.php:64 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/HttpKernel/CacheWarmer/CacheWarmerAggregate.php:52 { …}
      ./vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Command/CacheWarmupCommand.php:96 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:255 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:971 { …}
      ./vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:86 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:248 { …}
      ./vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:74 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:148 { …}
      ./app/console:27 {
        › $application = new Application($kernel);
        › $application->run($input);
        › 
      }
    }
  }
]

```

**Comment by greg0ire** [comment]
Consider upgrading to a maintained version of `doctrine/persistence`, it might be possible.

The piece of code that causes an issue is https://github.com/doctrine/persistence/blob/7a6eac9fb6f61bba91328f15aa7547f4806ca288/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php#L177-L183

Knowing what the value of `$className` is when the crash happens should help you troubleshoot this, and you could also dump what is passed in the foreach.

**Comment by AntoineLemaire** [comment]
After few tests, the issue has been fixed in Symfony 3.6.36 by https://github.com/symfony/symfony/commit/a94e76a866c15db5521720aa111cf194cccb4606

So I think we can close this issue.
